src/cargo-fmt/main.rs

```diff
@@ -110,16 +122,53 @@ fn execute() -> i32 {
         handle_command_status(format_crate(
             verbosity,
             &strategy,
-            opts.rustfmt_options,
+            rustfmt_args,
             Some(&manifest_path),
         ))
     } else {
-        handle_command_status(format_crate(
-            verbosity,
-            &strategy,
-            opts.rustfmt_options,
-            None,
-        ))
+        handle_command_status(format_crate(verbosity, &strategy, rustfmt_args, None))
+    }
+}
+
+fn convert_message_format_to_rustfmt_args(
+    message_format: String,
+    rustfmt_args: &mut Vec<String>,
+) -> Result<(), String> {
+    let contains_emit_mode = rustfmt_args.contains(&String::from("--emit"));
```

Comment:
We do not need to create a `String` and then take its reference, just passing the string literal will do :)

Reply:
That's what I was originally trying to do, but the compiler was not having it. Combination of `rustfmt_args` containing `String`s and what `contains` wanted 🤷‍♂ 

I'll have another go at it

Reply:
Btw I believe https://github.com/rust-lang/rust/issues/42671 is the same issue I was running into while trying to use the string literal.

I switched this to use the workaround described on that thread

Reply:
Ah, I have forgotten about this...I remember hitting this a while ago myself. Sorry!